### ISSUE TITLE
W1b: three rounding policies for fractional box coordinates — `bbox_to_mask` (inclusive on floats), `Boxes.to_mask` (round, half-open), `bbox_to_mask3d` (truncate)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4015]
All three mask builders agree on integer vertices and disagree on fractional ones, because each has its own undocumented rounding policy: `bbox_to_mask` compares the raw floats inclusively (`xx >= x_min & xx <= x_max`), `Boxes.to_mask` exports the exclusive `xyxy` box, `.round()`s it and slices half-open, and `bbox_to_mask3d` `.long()`-truncates. Boxes come out of `transform_boxes` with fractional coordinates as a matter of course.

## Repro

```python
import torch
from kornia.geometry.bbox import bbox_to_mask, bbox_to_mask3d
from kornia.geometry.boxes import Boxes

v = torch.tensor([[[1., 1.], [4., 1.], [4., 3.], [1., 3.]]])
f = torch.tensor([[[1.4, 1.4], [3.6, 1.4], [3.6, 2.6], [1.4, 2.6]]])
print("integer box:    bbox_to_mask", bbox_to_mask(v, 6, 6).sum().item(), " Boxes.to_mask", Boxes(v).to_mask(6, 6).sum().item())
print("fractional box: bbox_to_mask", bbox_to_mask(f, 6, 6).sum().item(), " Boxes.to_mask", Boxes(f).to_mask(6, 6).sum().item())
f3 = torch.tensor([[[1.4, 1.4, 1.4], [3.6, 1.4, 1.4], [3.6, 2.6, 1.4], [1.4, 2.6, 1.4],
                    [1.4, 1.4, 2.6], [3.6, 1.4, 2.6], [3.6, 2.6, 2.6], [1.4, 2.6, 2.6]]])
print("fractional 3D:  bbox_to_mask3d", bbox_to_mask3d(f3, (6, 6, 6)).sum().item())
```

## Observed

```
integer box:    bbox_to_mask 12.0  Boxes.to_mask 12.0
fractional box: bbox_to_mask 2.0   Boxes.to_mask 12.0      <- 6x apart for the same box
fractional 3D:  bbox_to_mask3d 12.0                        <- z 1..2, y 1..2, x 1..3 after truncation
```

For the fractional 2D box: `bbox_to_mask` keeps pixels `x in {2, 3}, y in {2}` (2 px); `Boxes.to_mask` rounds the exclusive export `(1.4, 1.4, 4.6, 3.6)` to `(1, 1, 5, 4)` and keeps `x in 1..4, y in 1..3` (12 px).

## Expected

One documented rasterization rule for fractional coordinates (the audit's proposal: round to nearest, then the inclusive/half-open rule that matches the module's arithmetic), shared by the three builders. Any choice changes at least two of them → window; meanwhile each convention block states its own rule.

## Classification

**Semantic contract.** Idiom 5 — a precondition (integer coordinates) that is documented nowhere and validated nowhere, so each builder silently applies its own.

## Disposition

**MERGE IN WINDOW — behavior change.** Canonical: one rasterization rule, decided once (see Expected). Shape: shared implementation, all three names kept. Divergence today (executed): `2.0` vs `12.0` set pixels for the same fractional box. This is the behavior-change discussion CONTRIBUTING asks for, and the decision is: **in the coordinated Repair & Deprecation window, not before. A PR that aliases or consolidates these ahead of it will be closed and pointed here.**

## Environment

kornia 0.9.0rc1 @ `0685b1f5` (main), torch 2.9.1, python 3.11, cpu. Found during the bbox/boxes conventions audit (batch 4, tracking issue #4020); the batch documents the behavior below as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5).
